### PR TITLE
Adding timeout for curl requests

### PR DESCRIPTION
**Pull request by hkousha**
### Summary
This PR adds timeouts to the curl request. This is important specially for the agent engine data.

### Implementation details
Adding timeout to the curl requst.
```
curl --max-time 900
```

### Testing

- [x] Works properly on Amazon Linux
- [x] Works properly on RHEL 7
- [ ] Works properly on Debian 8
- [ ] Works properly on Ubuntu 14.04

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License: yes

### CODE REVIEW

**Review by nmeyerhans — APPROVED** [review]
LGTM :shipit:

File: ecs-logs-collector.sh [hunk 1]
```diff
@@ -167,7 +167,7 @@ try_set_instance_infodir() {
   try "resolve instance-id"
 
   if command -v curl > /dev/null; then
-    instance_id=$(curl -s -m 3 http://169.254.169.254/latest/meta-data/instance-id 2>/dev/null)
+    instance_id=$(curl --max-time 900 -s -m 3 http://169.254.169.254/latest/meta-data/instance-id 2>/dev/null)
```
Comment (jahkeup) [inline ecs-logs-collector.sh]: The `-m` here (shortened version of `--max-time`, at least on my machine) should add a timeout, albeit much shorter than `900` seconds at a sensitive `3` seconds. I would prefer we use the `--long-option` form so let's drop the `-m` in favor of your `--max-time` :+1:

Are you seeing cases where the command got close to `900` seconds elapsed? If not, what do you think about `60` instead so as to not keep users on the edge of their seats? (Same for below change curl)

**Review by jahkeup — CHANGES_REQUESTED** [review]
Thanks for your contribution @hkousha ! I have a quick question and one minor fix to ask of you.

File: ecs-logs-collector.sh (hunk 1 above)
Comment (hkousha) [reply ecs-logs-collector.sh]: sounds good, I will get rid of `-m` and change the timeout to 3.

Comment (hkousha) [reply ecs-logs-collector.sh]: @jahkeup fixed

**Review by jahkeup — APPROVED** [review]
LGTM! :boat:

**Pull request merged**